src/deroff.rs

```diff
@@ -55,7 +55,7 @@ impl Deroffer {
     fn new() -> Deroffer {
         Deroffer {
             g_re_word: crate::regex!(r##"[a-zA-Z_]+"##),
-            g_re_number: crate::regex!(r##"[+-]?\d+"##),
+            g_re_number: crate::regex!(r##"^[+-]?\d+"##),
```

Comment:
Why do we need the `^` change? The original version uses the same as the current one. https://github.com/fish-shell/fish-shell/blob/master/share/tools/deroff.py#L268

Reply:
I think that it has to have the anchor based on the definition of Python's [`re.Pattern.match`](https://docs.python.org/3/library/re.html#re.Pattern.match), which unintuitively requires a match at the start of the string.

Reply:
Yes, that's it. That, and the correct functioning of the parser seems to depend on this behavior of https://docs.python.org/3/library/re.html#re.Pattern.match. Let me know if you think otherwise.

Reply:
Oh, I did not know `match` is the *beginning of string*. Then `g_re_word`, `g_re_not_backslash_or_whitespace` and `g_re_font` also needs some changes.